Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -545,7 +549,16 @@ public override GetContentHashListResult GetContentHashList(OperationContext con
                         {
                             if (store.TryGetValue(key, out var data, nameof(Columns.Metadata)))
                             {
-                                result = DeserializeContentHashList(data);
+                                var metadata = DeserializeMetadataEntry(data);
+                                result = metadata.ContentHashListWithDeterminism;
+
+                                // Update the time, only if no one else has changed it in the mean time. We don't
+                                // really care if this succeeds or not, because if it doesn't it only means someone
+                                // else changed the stored value before this operation but after it was read.
+#pragma warning disable EPC13 // Suspiciously unobserved result.
+                                CompareExchange(context, strongFingerprint, metadata.ContentHashListWithDeterminism, metadata.ContentHashListWithDeterminism);
+#pragma warning restore EPC13 // Suspiciously unobserved result.
+
                                 // TODO(jubayard): since we are inside the ContentLocationDatabase, we can validate that all

```

Comment:
There are a few TODOs in this PR and near its changes. Are these noted elsewhere (Kanban, sticky notes, etc)?

Reply:
I keep these in my org-mode. I do plan on going through these and removing/addressing as necessary before considering this complete.

---
In reply to: [305444338](https://github.com/microsoft/BuildXL/pull/621#discussion_r305444338) [](ancestors = 305444338)